bundle/config/mutator/verify_cli_version_test.go

```diff
@@ -0,0 +1,170 @@
+	"github.com/stretchr/testify/require"
+)
+
+type testCase struct {
+	currentVersion string
+	constraint     string
+	expectedError  string
+}
+
+func TestVerifyCliVersion(t *testing.T) {
+	testCases := []testCase{
+		{
+			currentVersion: "0.0.1",
+		},
+		{
+			currentVersion: "0.0.1",
+			constraint:     "0.100.0",
+			expectedError:  "Databricks CLI version constraint not satisfied. Required: 0.100.0, current: 0.0.1",
+		},
+		{
+			currentVersion: "0.0.1",
+			constraint:     ">= 0.100.0",
+			expectedError:  "Databricks CLI version constraint not satisfied. Required: >= 0.100.0, current: 0.0.1",
+		},
+		{
+			currentVersion: "0.100.0",
+			constraint:     "0.100.0",
+		},
+		{
+			currentVersion: "0.100.1",
+			constraint:     "0.100.0",
+			expectedError:  "Databricks CLI version constraint not satisfied. Required: 0.100.0, current: 0.100.1",
+		},
+		{
+			currentVersion: "0.100.1",
+			constraint:     ">= 0.100.0",
+		},
+		{
+			currentVersion: "0.100.0",
+			constraint:     "<= 1.0.0",
+		},
+		{
+			currentVersion: "1.0.0",
+			constraint:     "<= 1.0.0",
+		},
+		{
+			currentVersion: "1.0.0",
+			constraint:     "<= 0.100.0",
+			expectedError:  "Databricks CLI version constraint not satisfied. Required: <= 0.100.0, current: 1.0.0",
+		},
+		{
+			currentVersion: "0.99.0",
+			constraint:     ">= 0.100.0, <= 0.100.2",
+			expectedError:  "Databricks CLI version constraint not satisfied. Required: >= 0.100.0, <= 0.100.2, current: 0.99.0",
+		},
+		{
+			currentVersion: "0.100.0",
+			constraint:     ">= 0.100.0, <= 0.100.2",
+		},
+		{
+			currentVersion: "0.100.1",
+			constraint:     ">= 0.100.0, <= 0.100.2",
+		},
+		{
+			currentVersion: "0.100.2",
+			constraint:     ">= 0.100.0, <= 0.100.2",
+		},
+		{
+			currentVersion: "0.101.0",
+			constraint:     ">= 0.100.0, <= 0.100.2",
+			expectedError:  "Databricks CLI version constraint not satisfied. Required: >= 0.100.0, <= 0.100.2, current: 0.101.0",
+		},
+		{
+			currentVersion: "0.100.0-beta",
+			constraint:     ">= 0.100.0, <= 0.100.2",
+			expectedError:  "Databricks CLI version constraint not satisfied. Required: >= 0.100.0, <= 0.100.2, current: 0.100.0-beta",
+		},
+		{
+			currentVersion: "0.100.0-beta",
+			constraint:     ">= 0.100.0-0, <= 0.100.2-0",
+		},
+		{
+			currentVersion: "0.100.1-beta",
+			constraint:     ">= 0.100.0-0, <= 0.100.2-0",
+		},
+		{
+			currentVersion: "0.100.3-beta",
+			constraint:     ">= 0.100.0, <= 0.100.2",
+			expectedError:  "Databricks CLI version constraint not satisfied. Required: >= 0.100.0, <= 0.100.2, current: 0.100.3-beta",
+		},
+		{
+			currentVersion: "0.100.123",
+			constraint:     "0.100.*",
+		},
+		{
+			currentVersion: "0.100.123",
+			constraint:     "^0.100",
+			expectedError:  "invalid version constraint \"^0.100\" specified. Please specify the version constraint in the format (>=) 0.0.0(, <= 1.0.0)",
+		},
+	}
+
+	t.Cleanup(func() {
+		build.SetBuildVersion(build.DefaultSemver)
```

Comment:
I was writing a comment about a context getter/setter for the build info until I realized this will be set to garbage for test binaries anyway. Could you include a comment here to say as much? It doesn't really matter what we configure this to when testing as long as it is a valid semver version.